tools/halide_image_io.h

```diff
@@ -812,7 +812,14 @@ bool buffer_is_compact_planar(ImageType &im) {
         return false;
     }
     for (int d = 1; d < im.dimensions(); ++d) {
-        if (im.dim(d-1).stride() >= im.dim(d).stride()) {
+        if (im.dim(d-1).stride() > im.dim(d).stride()) {
+            return false;
+        }
+        // strides can only match if both strides are 1
+        // (this can happen when artificially adding dimension(s), e.g.
+        // to write a .tmp file)
+        if (im.dim(d-1).stride() == im.dim(d).stride() &&
+            !(im.dim(d-1).extent() == 1 && im.dim(d).extent() == 1)) {
```

Comment:
I think only one of these needs to be stride 1, correct? Consider the following buffer dims (extent, stride):

    (10, 1),
    (1, 10),
    (3, 10),

This is a compact planar buffer that would fail this check because `im.dim(d).extent() != 1` (where d == 2).

I *think* we should only check that dim d - 1 has extent 1...

Reply:
Yep. Fixed.